### DISCUSSION TITLE
Support for Simultaneous Render Output to Multiple Textures

### DISCUSSION THREAD

**Discussion by killer-of-Plato** [gfx-rs/wgpu#6060] [Ideas]
Hello again,

Is there no fixed-function mechanism to save a render output to two textures simultaneously? I need to capture and display the output, and while I can wait until the next cycle to copy the texture from the swap-chain to the intermediate buffer, this would take place within the same sync block within which the swap-chain is again updated.

While I can submit the copy operation prior to the render pipeline, I cannot technically enforce that the former will complete before the latter does. I’m reasonably certain that it wouldn’t present an issue, but I really don’t know what all is going on under the hood to justify that confidence.

If there was a way to concurrently save the output to the swap-chain and a capture_texture buffer, then I would be at liberty to copy from the capture_texture without worrying about a swap-chain update.

Thanks!

**Comment by kpreid** [comment]
*   In your `@fragment` shader function, return a struct with two copies of the same color.

    ```wgsl
    struct FragmentOutput {
        @location(0) color1: vec4f,
        @location(1) color2: vec4f,
    }
    ```

*   Create a render pass with two `color_attachments`, one for each texture.

That should be all you need to render to two places.

That said, it would also definitely work reliably if you rendered to your `capture_texture`, then copied that to the surface texture. Everything submitted to the `Queue` executes (or appears to execute) in the order it was submitted.
